addon/model.js

```diff
@@ -259,6 +259,7 @@ export default class MegamorphicModel extends Ember.Object {
   }
 
   initBaseModel(baseModelName) {
+    // TODO document that we are dependent on store.push pushing in included first which has the base record data in it
```

Comment:
That's actually the beauty of `recordForId` - we are not dependent, although we will not have the data until it has been pushed, but this should be fine.